### PR TITLE
[#46] Setup Travis for HIRS Integration Tests

### LINKED ISSUE #46: Setup Integration Tests in Travis CI
Presently we have Style Checking, Static Analysis, Unit Testing and Project Packaging covered/performed by Travis CI to hedge against regressions. Travis CI still needs updated to perform some level of Integration Tests to ensure that at the very minimum it's guaranteed that Provisioning service is always functioning in supported environments between the ACA and the 1.2 and 2.0 Provisioners.

An initial glimpse suggests Docker Compose may be a useful tool to accomplish this.

### PR DESCRIPTION
**Pull request by apldev3**
This merge request allows for the ACA and HIRS TPM 2.0 Provisioner to be run in a Docker container. Additionally, it sets up the Integration Testing environment using Docker Compose. Additional optimizations to be performed in #67 include a potential merger of the Package and Integration Test phases in Travis and handling of the original TPM 1.2 Provisioner.

Closes #46

### CODE REVIEW

File: .ci/integration-tests/setup-tpm2provisioner.sh [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+echo "Creating Platform Cert for Container"
+PC_DIR=/var/hirs/pc_generation
+mkdir -p $PC_DIR
+/opt/paccor/scripts/allcomponents.sh > $PC_DIR/componentsFile
+/opt/paccor/scripts/referenceoptions.sh > $PC_DIR/optionsFile
+/opt/paccor/scripts/otherextensions.sh > $PC_DIR/extensionsFile
+/opt/paccor/bin/observer -c $PC_DIR/componentsFile -p $PC_DIR/optionsFile -e $ek_cert_der -f $PC_DIR/observerFile
+/opt/paccor/bin/signer -o $PC_DIR/observerFile -x $PC_DIR/extensionsFile -b 20180101 -a 20280101 -N $RANDOM -k /HIRS/.ci/integration-tests/certs/ca.key -P /HIRS/.ci/integration-tests/certs/ca.crt --pem -f $PC_DIR/$platform_cert
+
```
Comment (busaboy1340) [inline .ci/integration-tests/setup-tpm2provisioner.sh]: Think it's just an emulator issue, but you may get errors loading the the certs into NVRam. Checking and releasing it seems to fix the problem:

```suggestion
 	if tpm2_nvlist | grep -q 0x1c00002; then
	  tpm2_nvrelease -x 0x1c00002 -a 0x40000001 
	fi
```

File: .ci/integration-tests/setup-tpm2provisioner.sh [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+# specifies attribute value in publicInfo struct
+# [0x2000A = ownerread|ownerwrite|policywrite])
+size=$(cat $ek_cert_der | wc -c)
+echo "Define nvram location for ek cert of size $size"
+tpm2_nvdefine -x 0x1c00002 -a 0x40000001 -t 0x2000A -s $size
+
+# Load key into TPM nvram
+echo "Load ek cert into nvram"
+tpm2_nvwrite -x 0x1c00002 -a 0x40000001 $ek_cert_der
+
```
Comment (busaboy1340) [inline .ci/integration-tests/setup-tpm2provisioner.sh]: Think it's just an emulator issue, but you may get errors loading the the certs into NVRam. Checking and releasing it seems to fix the problem:

```suggestion
 	if tpm2_nvlist | grep -q 0x1c90000; then
	  tpm2_nvrelease -x 0x1c90000 -a 0x40000001 
	fi
```

File: .ci/integration-tests/setup-tpm2provisioner.sh (hunk 1 above)
Comment (busaboy1340) [reply .ci/integration-tests/setup-tpm2provisioner.sh]: A couple of us had this issue using the 2.0 Emulator on our linux boxes.
And I had the same problem when testing with the Docker images.
I figured it wouldn't be a problem with the fresh installs. Or maybe I'm just messing something up somehow...

Comment (apldev3) [reply .ci/integration-tests/setup-tpm2provisioner.sh]: Yeah I'm not sure that any of us over here ran into that issue, and I haven't done a deep-dive on all the changes you've made on your branch. It looked like you did a lot of cleaning and refinement of this initial cut, so it's possible something shifted in transit?

File: .ci/docker/Dockerfile.tpm2provisioner [hunk 3]
```diff
@@ -0,0 +1,14 @@
+FROM hirs/hirs-ci:centos7
+
+MAINTAINER apl.dev3@jhuapl.edu
+
+# Install packages for installing HIRS TPM2 Provisioner
+RUN yum -y update && yum clean all
+# TODO(apldev3): Remove vim-common if/when Paccor updates (Also update Paccor version below)
+RUN yum install -y tpm2-tools libcurl procps-ng vim-common wget dbus && yum clean all
```
Comment (busaboy1340) [inline .ci/docker/Dockerfile.tpm2provisioner]: Could you add the package **python-requests** to this line. It'll be needed to run the systems tests.

Comment (apldev3) [reply .ci/docker/Dockerfile.tpm2provisioner]: It's done.

File: .ci/integration-tests/setup-tpm2provisioner.sh (hunk 1 above)
Comment (apldev3) [reply .ci/integration-tests/setup-tpm2provisioner.sh]: Done.

File: .ci/integration-tests/setup-tpm2provisioner.sh (hunk 2 above)
Comment (apldev3) [reply .ci/integration-tests/setup-tpm2provisioner.sh]: Done.

**Comment by apldev3** [comment]
@busaboy1340 did you have anything further you needed done here, or is everything squared away on your end?

**Comment by busaboy1340** [comment]
Nothing else. I'm good.
